### ISSUE TITLE
Dragging a cell to the top makes the tableview shake

### ISSUE THREAD

**Issue by dyegos** [adamshin/SwiftReorder#21]
I have implemented the SwiftReorder and it's really cool. I'm facing an issue when I drag a cell from the bottom to up, but when the cell gets to the top, the whole tableview starts to shake as you can see in the video attached. is there a way to fix it?

[Sample.zip](https://github.com/adamshin/SwiftReorder/files/1707976/Sample.zip)

**Comment by tonistair** [comment]
First of all nice project :)... I have the have the same issue... For me it only seems to happen when the TableView has multiple sections 🤔

**Comment by dyegos** [comment]
I found more info about this issue. The shaking behaviour only happens on iOS 11.

**Comment by adamshin** [comment]
Yeah. Unfortunately iOS 11 seems to have introduced some ui quirks that cause shaking/jittering when dragging cells around.

I've been thinking about a possible rewrite of the library. It would implement drag-and-drop by displaying "dummy views" in place of the actual cells and headers. That way we'd have manual control over the visual position of each element, instead of relying on the UITableView cell insertion/deletion animations.

There's a branch called `image-proxy` with a basic implementation of that in place. It still needs a lot of work though...

**Comment by PiotrMaleszczuk** [comment]
Try with this guys ->
```
self.tableView.estimatedRowHeight = 0
self.tableView.estimatedSectionHeaderHeight = 0
self.tableView.estimatedSectionFooterHeight = 0
```

**Comment by tonistair** [comment]
Thank you very much! That did the trick for me as well! Much appreciated!

**Comment by SunilSpaceo** [comment]
I also notice that its happens only when you have multiple sections in a table. In my case if I try with single section then its working but its shake while using multiple section in that table

**Comment by costachung** [comment]
I had the same issue when dragging cells around while the TableView has multiple sections. The animation changes the bounds on iOS 11. 

In updateDestinationRow(),  removing the bounds animations fixes the problem.
```
       if #available(iOS 11.0, *) {
            tableView.performBatchUpdates({
                tableView.deleteRows(at: [context.destinationRow], with: .fade)
                tableView.insertRows(at: [newContext.destinationRow], with: .fade)
            }) { (finshed) in
            }
            tableView.layer.removeAnimation(forKey: "bounds.origin")
            tableView.layer.removeAnimation(forKey: "bounds.size")
        } else {
            // Fallback on earlier versions
            tableView.beginUpdates()
            tableView.deleteRows(at: [context.destinationRow], with: .fade)
            tableView.insertRows(at: [newContext.destinationRow], with: .fade)
            tableView.endUpdates()
        }
```

or simply,
`tableView.layer.removeAllAnimations()`

**Comment by shahchirag2110** [comment]
@dyegos have you find any solution? I have same problem but notice one more thing it will not drag down top two cells in my case.

**Comment by shahchirag2110** [comment]
@dyegos @adamshin i solved the issue by this line.
```

if #available(iOS 11.0, *) {
            self.tblCoin.contentInsetAdjustmentBehavior = UIScrollViewContentInsetAdjustmentBehavior.never
        }
```
Hope this help others

**Comment by m3g0byt3** [comment]
Have the same issue recently with SwiftReorder 6.0.2 on iOS 11/12, in my case it was arised after adding self-sizing UITV section header and footer views into table view, although standard self-sizing UITV cells works just fine in the same table view previously.

Fixed by setting:
```
self.tableView.estimatedSectionHeaderHeight = 0.0
self.tableView.estimatedSectionFooterHeight = 0.0
```

And by manual height calculation in appropriate delegate methods:
```
tableView(_:heightForHeaderInSection:)
tableView(_:heightForFooterInSection:)
```

Thanks @adamshin for great library.
